lib/inspec/profile.rb

```diff
@@ -211,6 +212,32 @@ def locked_dependencies
       @locked_dependencies ||= load_dependencies
     end
 
+    def lockfile_exists?
+      File.exist?(lockfile_path)
+    end
+
+    def lockfile_path
+      File.join(cwd, 'inspec.lock')
+    end
+
+    def cwd
+      File.directory?(@target) ? @target : nil
+    end
+
+    def lockfile
+      @lockfile ||= if lockfile_exists?
+                      Inspec::Lockfile.from_file(lockfile_path)
+                    else
+                      generate_lockfile
+                    end
+    end
+
+    def generate_lockfile(vendor_path = nil)
```

Comment:
I think it is already.  To render it would disk you'd have to do something like we do in the `vendor` function in cli.rb.